### DISCUSSION TITLE
403 Error keeps popping up

### DISCUSSION THREAD

**Discussion by tenderassist** [pocketbase/pocketbase#2954] [Q&A]
I made a web application using Next js and pocketbase and have been using it for a while now with no issues, but for some reason today every time I try to do something on the web app it keeps returning a 403 error message. It never did this before and I haven’t changed any code in my web app or any settings in pocketbase. Any assistance would be greatly appreciated.

**Comment by ganigeorgiev** [comment]
Without a code sample showing how do you initialize the PocketBase JS SDK and how do you interact with it, I'm not sure how to help.

There are no changes to the auth APIs in the last couple releases (both PocketBase and JS SDK), so I doubt that this is caused by a recent update.

**Reply by tenderassist** [reply]
I also tried adding  "const authData = await pb.admins.authWithPassword('test@example.com', '1234567890');" with my email and password 
but it doesn't seem to help

**Comment by tenderassist** [comment]
Sorry for not being more descriptive. So basically I can do something like update a field and it'll work once or twice, but every now and then it'll return a 403 error even when using the same information as input. Sometimes it works again after the 403 error message pops up so I'm lost on what could be wrong.

**Reply by ganigeorgiev** [reply]
I'm sorry, but once again without a code sample I don't think I can help you.

Please understand that I have literally no context about your project, nor how did you integrated PocketBase with Next.js.

**Reply by tenderassist** [reply]
Here an example of how I integrated pocketbase with next js in my projects.

I included this in the top of my code to connect with pocketbase: 

" const PocketBase = require("pocketbase/cjs");
const pb = new PocketBase("url");
pb.autoCancellation(false); "

and this is an example of me trying to add a field to one of the collections:

" try {
        const record = await pb.collection("specials").create(data);
        setfeedback("Successfuly added Special " + bnsnum);
      } catch (error) {
        window.alert("ERROR: Unable to add Special!");
        setfeedback("");
      } "

On this page a user just needs to provide a number in an input field and then it will be added to the collection. I'm a bit slow lol, so please let me know if you need to see more of the code or is this enough?

**Reply by ganigeorgiev** [reply]
Unfortunately the above code sample is not enough to understand what you are doing and where the error could originate.

- Are you sure that the correct collection API rules are used and nothing has changed?

- Why do you use the `cjs` bundle?

- Where the PocketBase request is executed - in the browser or server-side (_the `pb` instance seems server-side initializated but the call shows usage of `window.alert` so I guess is executed in the browser?_)?

- Is the `pb` client in the above authenticated as regular app user or admin?

- Are you using a global/shared PocketBase instance for all of your requests? If Yes, then **most of the time this would be incorrect and unsafe if the above is executed server-side** unless you authenticated as single admin the entire time and there is no external user triggered action that will change the `pb.authStore`.

There are so many questions that I really don't know how to help.
